pylint/checkers/typecheck.py

```diff
@@ -1548,6 +1548,12 @@ def visit_call(self, node: nodes.Call) -> None:
 
         # 2. Match the keyword arguments.
         for keyword in keyword_args:
+            # Skip if `keyword` is the same name as a positional-only-parameter
+            # and a `**kwargs` parameter exists.
+            if called.args.kwarg and keyword in [
+                arg.name for arg in called.args.posonlyargs
+            ]:
+                continue
```

Comment:
What do you think about a followup PR for 3.0 that emits a new message for this scenario? The code works, but it's pretty suspect (it's more likely a user made a mistake than a user is trying to call a signature that intentionally uses this weird semantics).

Reply:
In that case, we could leave the linked issue open until we create the "lower severity message" it asks for.

Reply:
I'm open to that. So for now I can use #Refs instead of #Closes in the fragment and we can separately address adding the lower-severity message in a follow-up.

Reply:
I think your last comment on the issue made me think you were leaning against the idea of adding a new message and were considering this primarily as a false positive @jacobtylerwalls but happy to leave the issue open anyway until we reach a conclusion.

Reply:
> I think your last comment on the issue made me think you were leaning against the idea of adding a new message and were considering this primarily as a false positive @jacobtylerwalls but happy to leave the issue open anyway until we reach a conclusion.

I was extraordinarily indecisive!

But on second thought, yeah, I think a new message would be good. This is such an edge case, though, it doesn't need to take a high priority.

Reply:
What would the new message be ? It feels like this should be a warning, the value in kwargs is misleading (is it going to be superseded by the one in args ? Frankly without reading the example and comments I would not know). Also what if the code does ``apple = kwargs.get("apple", Default apple")`` (like you probably expect if you're giving kwargs), isn't it a redundant keyword args anyway ?

Reply:
> What would the new message be ? It feels like this should be a warning, the value in kwargs is misleading (is it going to be superseded by the one in args ?

My idea was that it needs to be a lower severity message than redundant-keyword-arg because it's a valid language feature, as opposed to something that will raise TypeError when tried, like the other use cases for redundant-keyword-arg.

However, it's a very, very unlikely language feature, and more likely that someone just wrote a bad call to a function that _didn't_ have the kind of signature shown in the test cases. We could in that case emit a lower-severity info message that's like "hey, are you sure you're want to be doing this strange thing?"

Reply:
Maybe `hidden-kwarg`?

Reply:
``redundant-keyword-arg`` would describes both the error and the warning well imo. In fact ``redundant-keyword-arg`` describes the warning better, while the error would be better described by ``multiple-value-for-argument`` (Like the text for the type error raised by python in this case: `` got multiple values for argument 'x'``). Our message name need to be unique and retro-compatible so a possible solution would be to rename ``redundant-keyword-arg`` => ``multiple-value-for-argument``, and create ``redundant-keyword-argument``. That's convoluted and confusing in particular for ``redundant-keyword-arg`` / ``redundant-keyword-argument`` though. Maybe we can live with ``redundant-keyword-arg`` for both ?

Reply:
There's good reasons for either of the two approaches now mentioned.
How about we: Revert the code-change in this merge-request (i.e. we emit `redundant-keyword-arg` for this case) and merge the rest (the test-cases and the updated test output) which can be useful if we re-visit this to add a new low-severity message?

Reply:
We could also name the warning ``kwargs-hidden-by-positional-args`` or ``hidden-kwarg`` or something else (?) like @jacobtylerwalls suggested so there's no confusion between the two. (And we can rename the error anyway, or not)